src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,210 @@
+import React from 'react';
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
+import { Filter } from '../../types/Filter';
+import { Todo } from '../../types/Todo';
+import { TodoItem } from '../TodoItem';
+
+type Props = {
+  todos: Todo[]
+  selectedFilter: string
+  newTitleTodo: string
+  isAdding: boolean
+  userId: number | null
+  onSetIsError: React.Dispatch<React.SetStateAction<boolean>>
+  onSetTypeError: React.Dispatch<React.SetStateAction<string>>
+  toLoad:() => Promise<void>
+  isDeletedComplete: boolean
+};
+
+export const TodoList: React.FC<Props> = ({
+  todos,
+  selectedFilter,
+  newTitleTodo,
+  isAdding,
+  userId,
+  onSetIsError,
+  onSetTypeError,
+  toLoad,
+  isDeletedComplete,
+}) => {
+  const visibleTodos = todos.filter(todo => {
+    switch (selectedFilter) {
+      case Filter.COMPLETED:
+        return todo.completed;
+      case Filter.ACTIVE:
+        return !todo.completed;
+      case Filter.ALL:
+      default:
+        return todo;
+    }
+  });
+
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      <TransitionGroup>
+        {visibleTodos.map(todo => (
+          <CSSTransition
+            key={todo.id}
+            timeout={300}
+            classNames="item"
+          >
+            <TodoItem
+              todo={todo}
+              key={todo.id}
+              isAdding={false}
+              onSetIsError={onSetIsError}
+              onSetTypeError={onSetTypeError}
+              toLoad={toLoad}
+              isDeletedComplete={isDeletedComplete}
+            />
+          </CSSTransition>
+        ))}
+        {isAdding && userId
+      && (
+        <CSSTransition
+          key={0}
+          timeout={300}
+          classNames="temp-item"
+        >
+          <TodoItem
+            todo={{
+              id: 0,
+              userId,
+              title: newTitleTodo,
+              completed: false,
+            }}
+            isAdding={isAdding}
+            onSetIsError={onSetIsError}
+            onSetTypeError={onSetTypeError}
+            toLoad={toLoad}
+            isDeletedComplete={isDeletedComplete}
+          />
+        </CSSTransition>
+      )}
+
+        {/* <div data-cy="Todo" className="todo completed">
+        <label className="todo__status-label">
+          <input
+            data-cy="TodoStatus"
+            type="checkbox"
+            className="todo__status"
+            defaultChecked
+          />
+        </label>
```

Comment:
Remove unnecessary comments from your final commit

Reply:
of course! thanks a lot